app/controllers/campaigns_controller.rb

```diff
@@ -0,0 +1,50 @@
+class CampaignsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_campaign, only: [:edit, :update, :show, :destroy]
+
+  def new
+    @campaign = Campaign.new
+  end
+
+  def create
+    @campaign = Campaign.new(campaign_params)
+
+    if @campaign.save
+      redirect_to characters_path
+    else
+      render 'new'
+    end
+  end
+
+  def index
+    @campaigns = []
```

Comment:
A good approach or at least a possible one, would be:

```ruby
@characters = current_user.characters
campaign_ids = @characters.pluck(:campaign_id)
@campaigns = Campaign.find(campaign_ids)
```

Under `app/facades` you need to create a class that encapsulates the characters and campaigns:

```ruby
class CampaignAndCharacters
  def initialize(user)
     @user = user
  end

  def characters
    @characters ||= @user.characters
  end
...
end
```

So then the controller would look like:

```ruby
@characters_and_campaigns = CampaignAndCharacters.new(current_user)
```